components/omega/test/infra/ErrorTest.cpp

```diff
@@ -0,0 +1,157 @@
+// Main test driver
+int main(int argc, char **argv) {
+
+   int RetVal = 0;
+
+   MPI_Init(&argc, &argv);
+
+   MachEnv::init(MPI_COMM_WORLD);
+   MachEnv *DefEnv = OMEGA::MachEnv::getDefault();
+   initLogging(DefEnv);
+
+   LOG_INFO("------ Error handler test ------");
+   // Test default constructor and success check
+   Error ErrDefault;
+   if (ErrDefault.isSuccess()) {
+      LOG_INFO("Default error constructor: PASS");
+   } else {
+      LOG_ERROR("Default error constructor: FAIL");
+   }
+
+   // Test constructors with no message and boolean tests
+   Error ErrorTestSuccess2(ErrorCode::Success);
+   if (ErrorTestSuccess2.isSuccess() and !ErrorTestSuccess2.isFail()) {
+      LOG_INFO("No message error constructor success: PASS");
+   } else {
+      LOG_ERROR("No message error constructor success: FAIL");
+   }
+   Error ErrorTestWarn2(ErrorCode::Warn);
+   if (!ErrorTestWarn2.isSuccess() and !ErrorTestWarn2.isFail() and
+       ErrorTestWarn2.Code == ErrorCode::Warn) {
+      LOG_INFO("No message error constructor warn: PASS");
+   } else {
+      LOG_ERROR("No message error constructor warn: FAIL");
+   }
+   Error ErrorTestFail2(ErrorCode::Fail);
+   if (ErrorTestFail2.isFail() and !ErrorTestFail2.isSuccess()) {
+      LOG_INFO("No message error constructor fail: PASS");
+   } else {
+      LOG_ERROR("No message error constructor fail: FAIL");
+   }
+
+   // Test constructor with message
+   // Create a total error to accumulate codes and messages
+   std::string ExpectMsg =
+       "   [error] [ErrorTest.cpp:121] Initialized to fail String 1\n";
+   Error TotalError(ErrorCode::Fail, __LINE__, __FILE__,
+                    "Initialized to fail {} {}", "String", 1);
+   if (TotalError.isFail() and TotalError.Msg == ExpectMsg) {
+      LOG_INFO("Error constructor with message: PASS");
+   } else {
+      LOG_ERROR("Error constructor with message: FAIL");
+      LOG_ERROR(" Error constructor with message expected: {}", ExpectMsg);
+      LOG_ERROR(" Error constructor with message actual:   {}", TotalError.Msg);
+   }
+
+   // Test reset of error code
+   TotalError.reset();
+   if (TotalError.isSuccess()) {
+      LOG_INFO("Reset of error code: PASS");
+   } else {
+      ERROR_CHECK(TotalError, "Reset of error code: FAIL");
+   }
+
+   // Call routines to check error return functions
+   TotalError += returnErrorRoutine1();
+
+   // Test ERROR_CHECK for adding and printing error message
+   ERROR_CHECK(TotalError, "Encountered error in returnErrorRoutine1");
+
+   // Test ERROR_CHECK for adding warning and error message
+   ERROR_CHECK_WARN(TotalError, "Encountered error in returnErrorRoutine1");
+
+   // The above calls have reset the error so create a new failure
+   TotalError += Error(ErrorCode::Fail, __LINE__, __FILE__, "Error in driver");
+
+   // Test critical error
+   passError1(TotalError);
+
+   // Finalize environments
+   MPI_Finalize();
+
+   return RetVal;
```

Comment:
"RetVal is not set by the each tests in this unit-test, so the test always returns the initial value, or passError1(TotalError) exits the test early, preventing any change to the returned RetVal."

Reply:
Thanks, I forgot to add these for the non-fatal tests - but I also realized after seeing the other FAIL below that this test code is expected to fail because of the critical error abort at the end, so I need to find an alternative way of flagging the other test failures.

Reply:
Fixed with latest commit